### PR TITLE
This PR is the first one for code refactoring.

### PR DESCRIPTION
**Pull request by 0xstepit**
This PR:
- Refactor App removing cosmoscmd references.
- Remove simulation files
- Introduce a cleaner mock app.

### CODE REVIEW

File: x/slashrefund/keeper/deposit_test.go [hunk 1]
```diff
@@ -3,47 +3,19 @@ package keeper_test
...
-		items[i].Shares = sdk.ZeroDec()
-		keeper.SetDeposit(ctx, items[i])
-	}
-	return items, valAddr
-}
+// -------------------------------------------------------------------------------------------------
+// Test deposit
+// -------------------------------------------------------------------------------------------------
 
 func TestDepositGet(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/deposit_test.go]: Should be renamed in TestGetDeposit since the tested function is GetDeposit().

File: x/slashrefund/keeper/deposit_test.go [hunk 2]
```diff
@@ -52,9 +24,10 @@ func TestDepositGet(t *testing.T) {
 		require.Equal(t, deposit, rst)
 	}
 }
+
 func TestDepositRemove(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/deposit_test.go]: Should be renamed in TestRemoveDeposit since the tested function is RemoveDeposit().

File: x/slashrefund/keeper/deposit_test.go [hunk 3]
```diff
@@ -66,14 +39,87 @@ func TestDepositRemove(t *testing.T) {
 
 func TestDepositGetAll(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/deposit_test.go]: Should be renamed in TestGetAllDeposit since the tested function is GetAllDeposit().

File: x/slashrefund/keeper/deposit_test.go [hunk 4]
```diff
@@ -66,14 +39,87 @@ func TestDepositRemove(t *testing.T) {
...
+	items1, valAddr1 := testslashrefund.CreateNDepositForValidator(keeper, ctx, 10)
 	require.ElementsMatch(t, items0, keeper.GetValidatorDeposits(ctx, valAddr0))
 	require.ElementsMatch(t, items1, keeper.GetValidatorDeposits(ctx, valAddr1))
 }
+
+// -------------------------------------------------------------------------------------------------
+// Test deposit pool
+// -------------------------------------------------------------------------------------------------
+
+func TestPoolGetDeposit(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/deposit_test.go]: Should be renamed in TestGetDepositPool since the tested function is GetDepositPool().

File: x/slashrefund/keeper/msg_server_test.go [hunk 5]
```diff
@@ -1,19 +1,236 @@
...
+
+	app, ctx := testsuite.CreateTestApp(delAccs, valAccs, balances, false)
+
+	return app, ctx, delAccs, valAccs, depAccs
+}
+
+// -------------------------------------------------------------------------------------------------
+// Deposit
+// -------------------------------------------------------------------------------------------------
+func TestDepositValid(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/msg_server_test.go:59]: Should check which error is returned with require.ErrorIs(), in order to be sure of the actual msgServer behaviour.

File: x/slashrefund/keeper/msg_server_test.go [hunk 6]
```diff
@@ -1,19 +1,236 @@
...
+		Amount: sdk.NewCoin(types.DefaultAllowedTokens[0], sdk.NewInt(0)),
+	}
+	_, err := msgServer.Withdraw(ctx, msg)
+	require.Error(t, err)
+}
+
+// -------------------------------------------------------------------------------------------------
+// Claim 
+// -------------------------------------------------------------------------------------------------
+func TestClaimotValidatorBech32(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/msg_server_test.go]: Typo: TestClaimotValidatorBech32 -> TestClaimNotValidatorBech32

File: x/slashrefund/keeper/msg_server_test.go [hunk 7]
```diff
@@ -1,19 +1,236 @@
...
+	msg := &types.MsgDeposit{
+		DepositorAddress: depositors[0].String(),
+		ValidatorAddress: validators[0].String(),
+		Amount: sdk.NewCoin("mib", sdk.NewInt(1)),
+	}
+	_, err := msgServer.Deposit(ctx, msg)
+	require.Error(t, err)
+}
+
+func TestDepositZeroAmount(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/msg_server_test.go:133]: Should add also a test for the negative deposit amount.

File: x/slashrefund/keeper/msg_server_test.go [hunk 8]
```diff
@@ -1,19 +1,236 @@
...
+	msg := &types.MsgWithdraw{
+		DepositorAddress: depositors[0].String(),
+		ValidatorAddress: validators[0].String(),
+		Amount: sdk.NewCoin("mib", sdk.NewInt(1)),
+	}
+	_, err := msgServer.Withdraw(ctx, msg)
+	require.Error(t, err)
+}
+
+func TestWithdrawZeroAmount(t *testing.T) {
```
Comment (niilptr) [inline x/slashrefund/keeper/msg_server_test.go:193]: Should add also a test for the negative withdraw amount.

File: x/slashrefund/testslashrefund/testkeeper.go [hunk 9]
```diff
@@ -30,7 +30,11 @@ import (
 	tmdb "github.com/tendermint/tm-db"
 )
 
+// NewTestKeeper is used to build all the mocked requirements for testing the slash refund keeper.
```
Comment (niilptr) [inline x/slashrefund/testslashrefund/testkeeper.go]: We should be clearer about mock apps. Since in sdk documentation it is recommended to use mock entities mocked through mockgen, this description may be misleading.

File: x/slashrefund/keeper/deposit_test.go (hunk 1 above)
Comment (0xstepit) [reply x/slashrefund/keeper/deposit_test.go]: right

File: x/slashrefund/keeper/deposit_test.go (hunk 2 above)
Comment (0xstepit) [reply x/slashrefund/keeper/deposit_test.go]: right

File: x/slashrefund/keeper/deposit_test.go (hunk 3 above)
Comment (0xstepit) [reply x/slashrefund/keeper/deposit_test.go]: right

File: x/slashrefund/keeper/deposit_test.go (hunk 4 above)
Comment (0xstepit) [reply x/slashrefund/keeper/deposit_test.go]: right

File: x/slashrefund/keeper/msg_server_test.go (hunk 5 above)
Comment (0xstepit) [reply x/slashrefund/keeper/msg_server_test.go:59]: good point
